activerecord/lib/active_record/model_schema.rb

```diff
@@ -447,7 +500,14 @@ def load_schema
         end
 
         def load_schema!
-          @columns_hash = connection.schema_cache.columns_hash(table_name).except(*ignored_columns)
+          @columns_hash = connection.schema_cache.columns_hash(table_name)
+
+          unless define_attributes_from_schema
```

Comment:
This seems like the wrong place to handle things. We should not be modifying `columns_hash` based on this setting (other than perhaps not setting it at all). It seems like we should be skipping the loop down below instead.